packages/marshal/src/rankOrder.js

```diff
@@ -101,15 +101,16 @@ const memoOfSorted = new WeakMap();
 const comparatorMirrorImages = new WeakMap();
 
 /**
- * @param {RankCompare=} compareRemotables
- * An option to create a comparator in which an internal order is
- * assigned to remotables. This defaults to a comparator that
- * always returns `0`, meaning that all remotables are tied
- * for the same rank.
+ * @param {PartialCompare} [compareRemotables]
+ * A comparator for assigning an internal order to remotables.
+ * It defaults to a function that always returns `NaN`, meaning that all
+ * remotables are incomparable and should tie for the same rank without further
+ * refinement (e.g., not only are `r1` and `r2` tied, but so are `[r1, 0]`
+ * and `[r2, "x"]`).
  * @returns {RankComparatorKit}
  */
-export const makeComparatorKit = (compareRemotables = (_x, _y) => 0) => {
-  /** @type {RankCompare} */
+export const makeComparatorKit = (compareRemotables = (_x, _y) => NaN) => {
+  /** @type {PartialCompare} */
```

Comment:
Confirm or deny:

Previously:
- values from dispirit ranks are compared by the total order of ranks
- all values within atom ranks, e.g., numbers and strings, are mutually comparable
- all values within composite ranks, e.g., arrays, are mutually comparable provided
  - that their respective entries are all comparable, or
  - if they contain any respective pairs that are incomparable, those arrays have non-equivalent prefixes
  - and are otherwise incomparable
- within the rank of promises, all values are incomparable
- within the rank of remotables, all values are equivalent
- comparing incomparable values produces 0

After:
- within the rank of remotables, all values are incomparable
- comparing incomparable values produces NaN

Reply:
That seems correct, although I would rephrase a bit:

Previously:
* …
* within the rank of remotables, all values are equivalent <ins>in the default comparison (used by `compareRank`)</ins>
* <del>comparing incomparable values produces 0</del> <ins>there is no concept of incomparable values within a rank (but encountering an unacceptable pass style such as "error" or "promise" results in a thrown exception)</ins>

After:
* within the rank of remotables, all values are incomparable
* comparing incomparable values produces NaN <ins>in the inner comparator and short-circuits the outer comparator result to 0</ins>

Reply:
So, both before and after this change, comparing any pair of promises would by default return NaN?

Consequently, both before and after this change, the range of comparison was -1, 0, 1, and NaN?

Consequently, both before and after this change, rank compare is and was a partial order?

Reply:
I’m far enough along in my read to see with my own eyes that, both before and after this change:

- all exported comparators produce values in the range -1, 0, +1 _for the default comparator for remotables_.
- all promises are equivalent, not incoparable